### ISSUE TITLE
Torcs training

### ISSUE THREAD

**Issue by sanjeevk42** [only4hj/DeepRL#1]
Can you share the training details(training parameters, algorithm, total training steps etc) for torcs? 
Also did you have issues related to entropy becoming zero while training with a3c?

**Comment by only4hj** [comment]
You can see hyperparameters in the file below.
https://github.com/only4hj/DeepRL/blob/master/env/torcs/torcs_env.py
(See parameters with option args.drl == 'dqn')
Main algorithm is DDPG.
The train is done under 1,000 episodes (4 hours in my PC) when low features are used as the input directly instead of image pixels.

When I used a3c I have no experience to see zero entropy as you mentioned.
Did you add entropy loss?

**Comment by sanjeevkumar42** [comment]
Actually I was training torcs with a3c implementation (https://github.com/openai/universe-starter-agent) by openai. But it doesn't seem to work for torcs due to drop in entropy after few iteration. I thought you might have encountered similar issue with a3c.

Did you try training torcs with image data only on a3c?

**Comment by only4hj** [comment]
I have not tried using a3c for torcs.
I used only DDPG for torcs.

**Comment by sanjeevkumar42** [comment]
Did you try DDPG with image data only?

**Comment by only4hj** [comment]
Yes I did. It works fine with DDPG using only image input. 
With low features input the network outputs three actions, steering, acceleration and brake.
With only image input i made the network output one action, steering of the car to make traing simple.

**Comment by elistar** [comment]
Do you have any plan to publish the pre-trained models for torcs with low dimensional or high dimensional features?

**Comment by only4hj** [comment]
For now there is no plan to publish trained models.
You can install and train them by following the setup below.
https://github.com/only4hj/DeepRL/blob/master/env/torcs/INSTALL.md

**Comment by uhvardhan** [comment]
Hi! I have been getting an error. I was hoping if you could help with this. It keeps printing the same output. 

Timeout for client answer
Timeout for client answer

**Comment by hilbertletanger** [comment]
On Torcs's graphical interface, the vehicle does not move very much. After a certain period of time (about 10 minutes), the vehicle bumps into the roadside (I suspect this is because the road is inclined and the car slides to one side automatically).

I was wondering if you could offer me some possible ways to solve this problem. I would appreciate your reply.


> Hi! I have been getting an error. I was hoping if you could help with this. It keeps printing the same output.
> 
> Timeout for client answer
> Timeout for client answer

same problem.I

**Comment by hilbertletanger** [comment]
> Hi! I have been getting an error. I was hoping if you could help with this. It keeps printing the same output.
> 
> Timeout for client answer
> Timeout for client answer
Same problem
and on Torcs's graphical interface, the vehicle does not move very much. After a certain period of time (about 10 minutes), the vehicle bumps into the roadside (I suspect this is because the road is inclined and the car slides to one side automatically).

I was wondering if you could offer me some possible ways to solve this problem. I would appreciate your reply.
